completions/_umount.linux

```diff
@@ -49,24 +47,23 @@ _linux_fstab()
 
     # Read and unescape values into COMPREPLY
     local fs_spec fs_file fs_other
-    local ifs="$IFS"
     while read -r fs_spec fs_file fs_other; do
         if [[ $fs_spec == [#]* ]]; then continue; fi
         if [[ ${1-} == -L ]]; then
             local fs_label=${fs_spec/#LABEL=/}
             if [[ $fs_label != "$fs_spec" ]]; then
                 __linux_fstab_unescape fs_label
-                IFS=$'\0'
+                local IFS=$'\0'
                 COMPREPLY+=("$fs_label")
-                IFS=$ifs
+                _comp_unlocal
```

Comment:
missing IFS. By the way, I don't see what is the intent of the original code: Does anything change here in the behavior by setting `IFS=$'\0'`? Do you have an idea?

Reply:
There is some attempt at escaping quotes. The NULs are injected above in __linux_fstab_unescape.

Reply:
Hmm, I still don't understand why they are needed. I don't have an idea why `__linux_fstab_unescape` can be related to `IFS=$'\0'`. First of all, we cannot include NUL in the variable values in Bash. Next, even if we could include NUL in the variables, `COMPREPLY+=("$fs_label")` will add the value as *a single array element* regardless of `IFS` because it is quited by `"..."`.

I have checked the change history of the relevant code section.

- https://github.com/scop/bash-completion/commit/99b289474ead02cc8c629ceec1192ce9f0b4f84e#diff-a33e10856930590d5777b4de609f66b088b7477e930a200bb2439e83724ef328L51-R87
- https://github.com/scop/bash-completion/commit/1c4b461882c94febb39a550e0fe3065208139757#diff-a33e10856930590d5777b4de609f66b088b7477e930a200bb2439e83724ef328R55-R60
- https://github.com/scop/bash-completion/commit/2f61acd068a67e630dd28ce04157908398dd8798#diff-a33e10856930590d5777b4de609f66b088b7477e930a200bb2439e83724ef328R50-R53
- https://github.com/scop/bash-completion/commit/4375c4b94ea7bdd370cd771aeb3483f36a42bd9f#diff-a33e10856930590d5777b4de609f66b088b7477e930a200bb2439e83724ef328L50-R62

It seems the code tries to evaluate the octal sequences of `\ooo` and split the field with the resulting NULs (encoded as `\000` in the original data). But, if that is the original intent, the implementation is actually broken from the beginning. It has never worked. We need to re-implement it using `read -d ''` which can be used for the NUL-separated data.

Reply:
I have opened a PR at #690 for this, so let me close the conversation.